### ISSUE TITLE
Add-HPOVStoragePool issue with -ApplianceConnection

### ISSUE THREAD

**Issue by tknightblc** [HewlettPackard/POSH-HPOneView#85]
Been working with latest POSH library Release 2.0.296.0, all working well except Add-HPOVStoragePool which complains about -ApplianceConnection parameter whether I add it or not. See info below where I run command twice, once without -ApplianceConnection and once with - also included error output, version, info, etc. 

I can get this same command to work with 1.20 library w/o -ApplianceConnection, but not with 2.xx library - all other commands working well.

Any ideas?

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> Get-HPOVVersion

PowerShell Library Version : 2.0.296.0
Library Path               : C:\Windows\system32\WindowsPowerShell\v1.0\Modules\HPOneView.200

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $PSVersionTable

Name  : PSVersion
Value : 4.0

Name  : WSManStackVersion
Value : 3.0

Name  : SerializationVersion
Value : 1.1.0.1

Name  : CLRVersion
Value : 4.0.30319.42000

Name  : BuildVersion
Value : 6.3.9600.16406

Name  : PSCompatibleVersions
Value : {1.0, 2.0, 3.0, 4.0}

Name  : PSRemotingProtocolVersion
Value : 2.2

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $Connection1

ConnectionID Name                          UserName      AuthLoginDomain
------------ ----                          --------      ---------------
1            tk1-ov200-dcsdemo.advizex.net Administrator LOCAL          

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $StoragePoolImport = @("FST_CPG1","FST_CPG2","CPG_SSD")

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> Add-HPOVStoragePool -poolName $StoragepoolImport -StorageSystem ThreePAR7200-4310
Add-HPOVStoragePool : A parameter cannot be found that matches parameter name 'ApppianceConnection'.
At line:1 char:1
+ Add-HPOVStoragePool -poolName $StoragepoolImport -StorageSystem ThreePAR7200-431 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Add-HPOVStoragePool], ParameterBindingException
    + FullyQualifiedErrorId : NamedParameterNotFound,Add-HPOVStoragePool
 

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $Connection1

ConnectionID Name                          UserName      AuthLoginDomain
------------ ----                          --------      ---------------
1            tk1-ov200-dcsdemo.advizex.net Administrator LOCAL          

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> Add-HPOVStoragePool -poolName $StoragepoolImport -StorageSystem ThreePAR7200-4310 -ApplianceConnection $Connection1
Add-HPOVStoragePool : A parameter cannot be found that matches parameter name 'ApppianceConnection'.
At line:1 char:1
+ Add-HPOVStoragePool -poolName $StoragepoolImport -StorageSystem ThreePAR7200-431 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Add-HPOVStoragePool], ParameterBindingException
    + FullyQualifiedErrorId : NamedParameterNotFound,Add-HPOVStoragePool
 

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $error | % { Format-List -InputObject $_ * } 

PSMessageDetails      : 
Exception             : System.Management.Automation.ParameterBindingException: A parameter cannot be found that matches parameter name 'ApppianceConnection'.
                           at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
                           at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
TargetObject          : 
CategoryInfo          : InvalidArgument: (:) [Get-HPOVStorageSystem], ParameterBindingException
FullyQualifiedErrorId : NamedParameterNotFound,Get-HPOVStorageSystem
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Add-HPOVStoragePool<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\HPOneView.200\HPOneView.200.psm1: line 21150
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}

PSMessageDetails      : 
Exception             : System.Management.Automation.ParameterBindingException: A parameter cannot be found that matches parameter name 'ApppianceConnection'.
                           at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
                           at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
TargetObject          : 
CategoryInfo          : InvalidArgument: (:) [Get-HPOVStorageSystem], ParameterBindingException
FullyQualifiedErrorId : NamedParameterNotFound,Get-HPOVStorageSystem
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Add-HPOVStoragePool<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\HPOneView.200\HPOneView.200.psm1: line 21150
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}

PSMessageDetails      : 
Exception             : System.Management.Automation.ParameterBindingValidationException: Cannot validate argument on parameter 'Pool'. The argument is null or empty. 
                        Provide an argument that is not null or empty, and then try the command again. ---> System.Management.Automation.ValidationMetadataException: 
                        The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
                           at System.Management.Automation.ValidateNotNullOrEmptyAttribute.Validate(Object arguments, EngineIntrinsics engineIntrinsics)
                           at System.Management.Automation.ParameterBinderBase.BindParameter(CommandParameterInternal parameter, CompiledCommandParameter 
                        parameterMetadata, ParameterBindingFlags flags)
                           --- End of inner exception stack trace ---
                           at System.Management.Automation.Internal.PipelineProcessor.SynchronousExecuteEnumerate(Object input, Hashtable errorResults, Boolean 
                        enumerate)
                           at System.Management.Automation.PipelineOps.InvokePipeline(Object input, Boolean ignoreInput, CommandParameterInternal[][] pipeElements, 
                        CommandBaseAst[] pipeElementAsts, CommandRedirection[][] commandRedirections, FunctionContext funcContext)
                           at System.Management.Automation.Interpreter.ActionCallInstruction`6.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
TargetObject          : 
CategoryInfo          : InvalidData: (:) [Add-HPOVStoragePool], ParameterBindingValidationException
FullyQualifiedErrorId : ParameterArgumentValidationError,Add-HPOVStoragePool
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}

PSMessageDetails      : 
Exception             : System.Management.Automation.ParameterBindingValidationException: Cannot validate argument on parameter 'Pool'. The argument is null or empty. 
                        Provide an argument that is not null or empty, and then try the command again. ---> System.Management.Automation.ValidationMetadataException: 
                        The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
                           at System.Management.Automation.ValidateNotNullOrEmptyAttribute.Validate(Object arguments, EngineIntrinsics engineIntrinsics)
                           at System.Management.Automation.ParameterBinderBase.BindParameter(CommandParameterInternal parameter, CompiledCommandParameter 
                        parameterMetadata, ParameterBindingFlags flags)
                           --- End of inner exception stack trace ---
                           at System.Management.Automation.Internal.PipelineProcessor.SynchronousExecuteEnumerate(Object input, Hashtable errorResults, Boolean 
                        enumerate)
                           at System.Management.Automation.PipelineOps.InvokePipeline(Object input, Boolean ignoreInput, CommandParameterInternal[][] pipeElements, 
                        CommandBaseAst[] pipeElementAsts, CommandRedirection[][] commandRedirections, FunctionContext funcContext)
                           at System.Management.Automation.Interpreter.ActionCallInstruction`6.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
TargetObject          : 
CategoryInfo          : InvalidData: (:) [Add-HPOVStoragePool], ParameterBindingValidationException
FullyQualifiedErrorId : ParameterArgumentValidationError,Add-HPOVStoragePool
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy> $error | % { $_.ScriptStackTrace } 
at Add-HPOVStoragePool<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\HPOneView.200\HPOneView.200.psm1: line 21150
at <ScriptBlock>, <No file>: line 1
at Add-HPOVStoragePool<Process>, C:\Windows\system32\WindowsPowerShell\v1.0\Modules\HPOneView.200\HPOneView.200.psm1: line 21150
at <ScriptBlock>, <No file>: line 1
at <ScriptBlock>, <No file>: line 1
at <ScriptBlock>, <No file>: line 1

PS C:\Users\tknight\Desktop\OneView\HP OV 2.0 DCS\Toms PS scripts - 2.0 working copy>

**Comment by ChrisLynchHPE** [comment]
This is a small bug in the Cmdlet.  For now, please use this as a workaround or solution to adding Storage Pools:

```PowerShell
$StoragePoolImport = "FST_CPG1","FST_CPG2","CPG_SSD"
Get-HPOVStorageSystem ThreePAR7200-4310 | Add-HPOVStoragePool -poolName $StoragepoolImport
```

This will bypass the code that performs the Storage System resource lookup when you provide a Storage System Name.

**Comment by tknightblc** [comment]
OK thanks Chris will try this now. Will this be fixed in next release?

**Comment by ChrisLynchHPE** [comment]
Yes.  I found the mistake and fixed it.

**Comment by tknightblc** [comment]
Perfect thanks Chris

**Comment by ChrisLynchHPE** [comment]
Fixed in the [2.0.310.0] (https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v2.0.310.0) release.
